### PR TITLE
Add a network switcher

### PR DESCRIPTION
**Pull request by har777**

### CODE REVIEW

File: lib/hooks.ts [hunk 1]
```diff
@@ -0,0 +1,15 @@
+import { RefObject, useEffect } from "react";
+
+export const useOutsideClick = (ref: RefObject<HTMLElement>, callback: any) => {
+  useEffect(() => {
+    const handleClickOutside = (event: any) => {
+      if (ref.current && !ref.current.contains(event.target)) {
+        callback();
+      }
+    };
+    document.addEventListener("mousedown", handleClickOutside);
```
Comment (Riglanto) [inline lib/hooks.ts:10]: on click? otherwise it doesn't work on mobile I think? :thinking:

**Comment by gglucass** [comment]
@har777 thanks a lot for this! 

When I use the transfer function I still get the following error: 
image


Furthermore it would be good if we also add the chain switcher to the /impact page. I think this probably warrants a different issue though.

File: components/Modals/BuyingGuideModal.tsx [hunk 2]
```diff
@@ -12,10 +12,6 @@ import { sliceAddress } from "@/lib/utils";
 import { buyWithUniswap } from "@/payments";
 import { USDC_POLYGON_CONTRACT_ADDRESS } from "@/utils";
```
Comment (gglucass) [inline components/Modals/BuyingGuideModal.tsx:13]: @har777 we also need USDC_MAINNET_CONTRACT_ADDRESS variability here

Comment (har777) [reply components/Modals/BuyingGuideModal.tsx:13]: good catch!

Comment (Riglanto) [reply components/Modals/BuyingGuideModal.tsx:13]: also note that `USDC_POLYGON_CONTRACT_ADDRESS` works only on prod
I did this trick to default to MATIC on non-prod envs to anything is being displayed:
https://github.com/Glo-Foundation/glo-wallet/pull/305/files#diff-f54cb834b7fbc073729588ea3b3b774a25102a62640e8b5ffc8e03eeb6f1aa6eR39

**Review by gglucass — APPROVED** [review]

**Pull request merged**